src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -1,19 +1,90 @@
 package core.basesyntax.impl;
 
 import core.basesyntax.Storage;
+import java.lang.reflect.Array;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int ARRAY_LEN = 10;
+    @SuppressWarnings("unchecked")
+    private final Entry[] entries = (Entry[]) Array.newInstance(Entry.class, ARRAY_LEN);
+    private int size = 0;
+
+    public StorageImpl() {
+    }
+
     @Override
     public void put(K key, V value) {
+        Entry entry = findEntry(key);
+
+        if (entry == null) {
+            checkArraySize();
+            entries[size++] = new Entry(key, value);
+        } else {
+            entry.setValue(value);
+        }
     }
 
     @Override
     public V get(K key) {
+        for (int i = 0; i < size; i++) {
+            Entry entry = entries[i];
+
+            if (areKeysEqual(key, entry.getKey())) {
+                return entry.getValue();
+            }
+        }
+
         return null;
     }
 
     @Override
     public int size() {
-        return -1;
+        return size;
+    }
+
+    private Entry findEntry(K key) {
+        for (int i = 0; i < size; i++) {
+            Entry entry = entries[i];
+
+            if (areKeysEqual(key, entry.getKey())) {
+                return entry;
+            }
+        }
+
+        return null;
+    }
+
+    private void checkArraySize() {
+        if (size == ARRAY_LEN) {
+            throw new RuntimeException("Storage is full. Maximum capacity of " + ARRAY_LEN
+                    + " reached.");
+        }
+    }
+
+    private boolean areKeysEqual(K one, K other) {
+        return one == null && other == null
```

Comment:
May be a bit simplified
```suggestion
        return one == other
```